packages/backend/src/modules/tracked-txs/modules/liveness/indexers/AnomaliesIndexer.ts

```diff
@@ -0,0 +1,265 @@
+import { TrackedTxConfigEntry } from '@l2beat/shared'
+import {
+  assert,
+  ProjectId,
+  TrackedTxsConfigSubtype,
+  UnixTime,
+  clampRangeToDay,
+  notUndefined,
+} from '@l2beat/shared-pure'
+import { Project } from '../../../../../model/Project'
+import {
+  ManagedChildIndexer,
+  ManagedChildIndexerOptions,
+} from '../../../../../tools/uif/ManagedChildIndexer'
+import { RunningStatistics } from '../api/RollingVariance'
+import {
+  AnomaliesRecord,
+  AnomaliesRepository,
+} from '../repositories/AnomaliesRepository'
+import {
+  LivenessRecordWithSubtype,
+  LivenessRepository,
+} from '../repositories/LivenessRepository'
+import { Interval, calculateIntervals } from '../utils/calculateIntervals'
+import { getProjectsToSync } from '../utils/getProjectsToSync'
+import { groupByType } from '../utils/groupByType'
+
+export interface AnomaliesIndexerIndexerDeps
+  extends Omit<ManagedChildIndexerOptions, 'name'> {
+  livenessRepository: LivenessRepository
+  anomaliesRepository: AnomaliesRepository
+  projects: Project[]
+}
+
+export class AnomaliesIndexer extends ManagedChildIndexer {
+  private readonly SYNC_RANGE = 30
+  constructor(private readonly $: AnomaliesIndexerIndexerDeps) {
+    super({ ...$, name: 'anomalies' })
```

Comment:
anomalies it too generic name, can lead to overlap in the future

Reply:
Let's leave it as it is for know as this is exactly the type of object we are dealing with